spec/unit/indirector/catalog/json_spec.rb

```diff
@@ -1,12 +1,67 @@
 #! /usr/bin/env ruby
 require 'spec_helper'
+require 'puppet_spec/files'
 require 'puppet/resource/catalog'
 require 'puppet/indirector/catalog/json'
 
 describe Puppet::Resource::Catalog::Json do
-  # This is it for local functionality: we don't *do* anything else.
+  include PuppetSpec::Files
+
+  # This is it for local functionality
   it "should be registered with the catalog store indirection" do
     expect(Puppet::Resource::Catalog.indirection.terminus(:json)).
       to be_an_instance_of described_class
   end
+
+  describe "when handling requests" do
+    let(:binary) { "\xC0\xFF".force_encoding(Encoding::BINARY) }
+    let(:key)    { 'foo' }
+    let(:file)   { subject.path(key) }
+    let(:catalog) do
+      catalog = Puppet::Resource::Catalog.new(key, Puppet::Node::Environment.create(:testing, []))
+      catalog.add_resource(Puppet::Resource.new(:file, '/tmp/a_file', :parameters => { :content => binary }))
+      catalog
+    end
+
+    before :each do
+      Puppet.run_mode.stubs(:master?).returns(true)
+      Puppet[:server_datadir] = tmpdir('jsondir')
+      FileUtils.mkdir_p(File.join(Puppet[:server_datadir], 'indirector_testing'))
+    end
+
+    def with_content(text)
```

Comment:
It doesn't look like we use this method in the test - maybe from an earlier iteration?